From bb6f896ac4c51fb53bbc73a0515a875dc9f49c01 Mon Sep 17 00:00:00 2001
From: Bart Van Assche <bvanassche@acm.org>
Date: Thu, 26 Dec 2019 00:36:30 +0000
Subject: [PATCH] qla2x00t-32gbit: Add debug dump of LOGO payload and ELS IOCB

The change adds a way to debug LOGO ELS, likewise PLOGI.

Link: https://lore.kernel.org/r/20191125165702.1013-14-r.bolshakov@yadro.com
Acked-by: Himanshu Madhani <hmadhani@marvell.com>
Acked-by: Quinn Tran <qutran@marvell.com>
Reviewed-by: Hannes Reinecke <hare@suse.de>
Tested-by: Hannes Reinecke <hare@suse.de>
Reviewed-by: Bart Van Assche <bvanassche@acm.org>
Signed-off-by: Roman Bolshakov <r.bolshakov@yadro.com>
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>

[ commit 9545767db3cde2d4f9ca581d096d97ed00754282 upstream ]


git-svn-id: http://svn.code.sf.net/p/scst/svn/trunk@8741 d57e44dd-8a1f-0410-8b47-8ef2f437770f
---
 qla2x00t-32gbit/qla_iocb.c | 9 +++++++++
 1 file changed, 9 insertions(+)

Index: scst-3.4.x/qla2x00t-32gbit/qla_iocb.c
===================================================================
--- scst-3.4.x.orig/qla2x00t-32gbit/qla_iocb.c
+++ scst-3.4.x/qla2x00t-32gbit/qla_iocb.c
@@ -2631,6 +2631,10 @@ qla24xx_els_dcmd_iocb(scsi_qla_host_t *v
 
 	memcpy(elsio->u.els_logo.els_logo_pyld, &logo_pyld,
 	    sizeof(struct els_logo_payload));
+	ql_dbg(ql_dbg_disc + ql_dbg_buffer, vha, 0x3075, "LOGO buffer:");
+	ql_dump_buffer(ql_dbg_disc + ql_dbg_buffer, vha, 0x010a,
+		       elsio->u.els_logo.els_logo_pyld,
+		       sizeof(*elsio->u.els_logo.els_logo_pyld));
 
 	rval = qla2x00_start_sp(sp);
 	if (rval != QLA_SUCCESS) {
@@ -2702,6 +2706,11 @@ qla24xx_els_logo_iocb(srb_t *sp, struct
 		els_iocb->rx_byte_count = 0;
 		els_iocb->rx_address = 0;
 		els_iocb->rx_len = 0;
+		ql_dbg(ql_dbg_io + ql_dbg_buffer, vha, 0x3076,
+		       "LOGO ELS IOCB:");
+		ql_dump_buffer(ql_log_info, vha, 0x010b,
+			       els_iocb,
+			       sizeof(*els_iocb));
 	}
 
 	sp->vha->qla_stats.control_requests++;
